agent/containermetadata/write_metadata_unix.go

```diff
@@ -19,23 +19,38 @@ import (
 	"fmt"
 	"path/filepath"
 
+	"github.com/cihub/seelog"
+
 	"github.com/aws/amazon-ecs-agent/agent/utils/ioutilwrapper"
 	"github.com/aws/amazon-ecs-agent/agent/utils/oswrapper"
 
 	"github.com/pborman/uuid"
 )
 
 const (
-	mountPoint = "/opt/ecs/metadata"
-	tempFile   = "temp_metadata_file"
+	mountPoint            = "/opt/ecs/metadata"
+	tempFile              = "temp_metadata_file"
+	bindMode              = "Z"
+	selinuxSecurityOption = "selinux"
 )
 
 // createBindsEnv will do the appropriate formatting to add a new mount in a container's HostConfig
 // and add the metadata file path as an environment variable ECS_CONTAINER_METADATA_FILE
 // We add an additional uuid to the path to ensure it does not conflict with user mounts
-func createBindsEnv(binds []string, env []string, dataDirOnHost string, metadataDirectoryPath string) ([]string, []string) {
+func createBindsEnv(binds []string, env []string, dataDirOnHost string, metadataDirectoryPath string, dockerSecurityOptions []string) ([]string, []string) {
+	selinuxEnabled := false
+	for _, option := range dockerSecurityOptions {
+		if option == selinuxSecurityOption {
+			selinuxEnabled = true
+		}
+	}
+
 	randID := uuid.New()
 	instanceBind := fmt.Sprintf(`%s/%s:%s/%s`, dataDirOnHost, metadataDirectoryPath, mountPoint, randID)
+	if selinuxEnabled {
+		seelog.Infof("Selinux is enabled on docker, mounting data directory in Z mode")
+		instanceBind = instanceBind + ":" + bindMode
```

Comment:
nit: Can use `fmt.sprintf()` like in next line.

Reply:
Done.